pkg/agent/multicluster/mc_route_controller_test.go

```diff
@@ -290,3 +426,177 @@ func TestMCRouteControllerAsRegularNode(t *testing.T) {
 	case <-finishCh:
 	}
 }
+
+func TestRemoveWireGuardRouteAndPeer(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockInterface := routemock.NewMockInterface(ctrl)
+	networkConfig := &config.NetworkConfig{}
+	c := newMCDefaultRouteController(t,
+		&config.NodeConfig{
+			Name: "node-4",
+			PodIPv4CIDR: &net.IPNet{
+				IP: net.ParseIP("10.10.0.0/16"),
+			},
+		},
+		networkConfig,
+		agent.WireGuardConfig{},
+		mockInterface,
+		"wireGuard",
+	)
+	c.wireGuardClient = &wireguardmock.MockWireGuardClient{}
+	defer c.queue.ShutDown()
+	mockInterface.EXPECT().DeleteRouteForLink(gomock.Any(), gomock.Any()).Times(1)
+
+	testCases := []struct {
+		name        string
+		ciImport    *mcv1alpha1.ClusterInfoImport
+		expectError error
+	}{
+		{
+			name: "remove peer successfully",
+			ciImport: &mcv1alpha1.ClusterInfoImport{
+				Spec: mcv1alpha1.ClusterInfo{
+					ServiceCIDR: "10.100.0.0/16",
+				},
+			},
+		},
+	}
+
+	for _, tt := range testCases {
```

Comment:
You should run test or assert inside of below block, so you can learn which one is failed clearer when there is any error.
```
t.Run(tt.name, func(t *testing.T) {
  assert.Equal(t, tt.expectError, c.removeWireGuardRouteAndPeer(tt.ciImport))
}
```

Reply:
Done. As well as other tests.